### DISCUSSION TITLE
apollo 如何注册到consul中，请问有案例教程吗？

### DISCUSSION THREAD

**Discussion by nongmingdashen** [apolloconfig/apollo#3555] [Q&A] [answered]
apollo 如何注册到consul中，请问有案例教程吗？

**Comment by Anilople** [comment]
目前还不行，代码上还不支持

**Reply by nongmingdashen** [reply]
我查资料 说是可以参照默认 meta service目录下的默认实现，增加一个consul的实现    难道不行吗？

**Reply by Anilople** [reply]
嗯，所以需要改动代码

目录在 https://github.com/ctripcorp/apollo/tree/master/apollo-configservice/src/main/java/com/ctrip/framework/apollo/metaservice/service

需要加入一个类似`ConsulDiscoveryService`之类的实现后，才可以支持注册到Consul

目前是没法通过改配置注册到Consul的哈

**Reply by nongmingdashen** [reply]
请问一下 有案例吗？   不知可以提供一下案例吗？

**Reply by Anilople** [reply]
issue已提 #3556

等待实现

**Reply by nongmingdashen** [reply]
好的 谢谢

**Comment by Anilople** [comment, marked as answer]
已经在 #3575 中得到实现
Implemented in  #3575
